compiler-core/templates/gleam@@compile.erl

```diff
@@ -43,20 +66,83 @@ worker_loop(Parent, Out) ->
     Options = [report_errors, report_warnings, debug_info, {outdir, Out}],
     erlang:send(Parent, {work_please, self()}),
     receive
-        {module, Module} -> 
+        {module, Module} ->
             log({compiling, Module}),
             case compile:file(Module, Options) of
-                {ok, _} -> 
-                    log({compiled, Module});
-                error -> 
+                {ok, ModuleName} ->
+                    Beam = filename:join(Out, ModuleName) ++ ".beam",
+                    Message = {compiled, Beam},
+                    log(Message),
+                    erlang:send(Parent, Message);
+                error ->
                     log({failed, Module}),
                     erlang:send(Parent, failed)
             end,
             worker_loop(Parent, Out)
     end.
 
+compile_elixir(Modules, Out) ->
+    Error = [
+        "The program elixir was not found. Is it installed?",
+        $\n,
+        "Documentation for installing Elixir can be viewed here:",
+        $\n,
+        "https://gleam.run/getting-started/"
+    ],
+    case Modules of
+        [] -> {true, []};
+        _ ->
+            log({starting, "compiler.app"}),
+            ok = application:start(compiler),
+            log({starting, "elixir.app"}),
+            case application:start(elixir) of
+                ok -> do_compile_elixir(Modules, Out);
+                _ ->
+                    io:put_chars([Error, $\n]),
+                    {false, []}
+            end
+    end.
+
+do_compile_elixir(Modules, Out) ->
+    ModuleBins = lists:map(fun(Module) ->
+        log({compiling, Module}),
+        list_to_binary(Module)
+    end, Modules),
+    OutBin = list_to_binary(Out),
+    Options = [{dest, OutBin}],
+    ShouldLog = should_log(),
+    % Silence "redefining module" warnings
+    'Elixir.Code':compiler_options([{ignore_module_conflict, true}]),
+    case 'Elixir.Kernel.ParallelCompiler':compile_to_path(ModuleBins, OutBin, Options) of
+        {ok, ModuleAtoms, _} ->
+            ToBeam = fun(ModuleAtom) ->
+                Beam = filename:join(Out, atom_to_list(ModuleAtom)) ++ ".beam",
+                log({compiled, Beam}),
+                Beam
+            end,
+            {true, lists:map(ToBeam, ModuleAtoms)};
+        {error, Errors, _} when ShouldLog ->
```

Comment:
I think we can remove this guard because it already is called within the log function. All other code can always log and not worry about whether or not it should actually be printed